src/test/java/uk/gov/hmcts/reform/jobscheduler/jobs/HttpCallJobTest.java

```diff
@@ -49,21 +53,27 @@ public void execute_should_send_http_request_specified_in_job_details() {
         given(context.getJobDetail())
             .willReturn(newJob(HttpCallJob.class).withIdentity("id", "group").build());
 
+        Map<String, String> headers = new HashMap<>();
+        headers.put("X-Custom-Header", "anything");
+        headers.put("ServiceAuthorization", "some-token");
+
         given(actionExtractor.extract(context))
             .willReturn(new HttpAction(
                 "http://localhost:8080/hello-world",
                 HttpMethod.POST,
-                ImmutableMap.of("ServiceAuthorization", "some-token"),
+                headers,
                 "some-body"
             ));
 
+        given(authTokenGenerator.generate()).willReturn("newly-generated-token");
         // when
-        new HttpCallJob(restTemplate, actionExtractor).execute(context);
+        new HttpCallJob(restTemplate, actionExtractor, authTokenGenerator).execute(context);
 
         // then
         verify(
             postRequestedFor(urlEqualTo("/hello-world"))
-                .withHeader("ServiceAuthorization", equalTo("some-token"))
+                .withHeader("X-Custom-Header", equalTo("anything"))
+                .withHeader("ServiceAuthorization", equalTo("newly-generated-token"))
```

Comment:
Now this single test is not only about 'should_send_http_request_specified_in_job_details', but also:
- `ServiceAuthorization` header is added
- `ServiceAuthorization` header if replaced if it already existed in action specified in job details

Can we have multiple tests? If not, perhaps just update the test method name so that it's clear what it's actually testing

Reply:
Thanks, renamed the test.

Reply:
You've only removed 'execute_' from test name (which was the name of the method that's being tested)

Reply:
I agree, it would be better if there were tests focused on auth header alone (and not caring about other details of the request). This way we would have very granular and distinctive cases - adds auth header when absent, replaces when present.

Reply:
Refactored to more descriptive test names.